pkg/instrumentation/auto/config.go

```diff
@@ -29,6 +37,70 @@ func (c AnnotationConfig) getResources(instType instrumentation.Type) Annotation
 	}
 }
 
+// LanguagesOf get languages to annotate for an object
+func (c AnnotationConfig) LanguagesOf(obj client.Object, checkNamespace bool) instrumentation.TypeSet {
+	objName := namespacedName(obj)
+	typesSelected := instrumentation.TypeSet{}
+
+	types := instrumentation.SupportedTypes
+
+	if checkNamespace {
+		for t := range types {
+			if slices.Contains(c.getResources(t).Namespaces, obj.GetNamespace()) {
+				typesSelected[t] = nil
+			}
+		}
+	}
+
+	switch obj.(type) {
+	case *appsv1.Deployment:
+		for t := range types {
+			if slices.Contains(c.getResources(t).Deployments, objName) {
+				typesSelected[t] = nil
+			}
+		}
+	case *appsv1.StatefulSet:
+		for t := range types {
+			if slices.Contains(c.getResources(t).StatefulSets, objName) {
+				typesSelected[t] = nil
+			}
+		}
+	case *appsv1.DaemonSet:
+		for t := range types {
+			if slices.Contains(c.getResources(t).DaemonSets, objName) {
+				typesSelected[t] = nil
+			}
+		}
+	case *corev1.Namespace:
+		for t := range types {
+			if slices.Contains(c.getResources(t).Namespaces, objName) {
+				typesSelected[t] = nil
+			}
+		}
+	}
+
+	return typesSelected
+}
+
+func (c AnnotationConfig) Empty() bool {
```

Comment:
super minor nit: Naming might suggest that it would empty the configuration.
```suggestion
func (c AnnotationConfig) IsEmpty() bool {
```